### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by bjlang**
# nf-core/atacseq pull request

Many thanks for contributing to nf-core/atacseq!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If necessary, also make a PR on the [nf-core/atacseq branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/atacseq)
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Documentation in `docs` is updated
- [ ] `CHANGELOG.md` is updated
- [ ] `README.md` is updated

**Learn more about contributing:** [CONTRIBUTING.md](https://github.com/nf-core/atacseq/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by bjlang** [comment]
@nf-core-bot fix linting

File: bin/check_samplesheet.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def check_samplesheet(file_in, file_out):
                         )
 
             ## Check replicate column is integer
-            if not replicate.isdigit():
+            if not replicate.isdecimal():
```
Comment (drpatelh) [inline bin/check_samplesheet.py:101]: Why this change? We are expecting an integer right?

File: bin/check_samplesheet.py [hunk 2]
```diff
@@ -137,14 +137,24 @@ def check_samplesheet(file_in, file_out):
 
                 ## Check that replicate ids are in format 1..<num_replicates>
                 uniq_rep_ids = sorted(list(set(sample_mapping_dict[sample].keys())))
-                if len(uniq_rep_ids) != max(uniq_rep_ids):
+                if len(uniq_rep_ids) != max(uniq_rep_ids) or 1 != min(uniq_rep_ids):
```
Comment (drpatelh) [inline bin/check_samplesheet.py:140]: 👀 What does this fix?

File: bin/check_samplesheet.py [hunk 3]
```diff
@@ -137,14 +137,24 @@ def check_samplesheet(file_in, file_out):
...
                     print_error(
                         "Replicate ids must start with 1..<num_replicates>!",
                         "Sample",
                         "{}, replicate ids: {}".format(sample, ",".join([str(x) for x in uniq_rep_ids])),
                     )
                     sys.exit(1)
 
+                ## Check that multiple runs of the same sample are of the same datatype i.e. single-end / paired-end
```
Comment (drpatelh) [inline bin/check_samplesheet.py:148]: This is already on `dev` no? 
https://github.com/nf-core/atacseq/blob/e05b4fb0f0ca8b84d78802c3b413ad81d45456ef/bin/check_samplesheet.py#L149-L159

**Review by drpatelh — COMMENTED** [review]
Thanks @bjlang ! Just need some clarification regarding the proposed changes here.

File: bin/check_samplesheet.py (hunk 2 above)
Comment (JoseEspinosa) [reply bin/check_samplesheet.py:140]: I think is checking whether the first replicate is not named as 1 but as any other number, then it won't fit in 1..<num_replicates>

Comment (bjlang) [reply bin/check_samplesheet.py:140]: It prevents starting with 0 and then skipping one number, i.e 0,2,3

File: bin/check_samplesheet.py (hunk 3 above)
Comment (bjlang) [reply bin/check_samplesheet.py:148]: The introductory comment line is wrong and should be something like "Check that multiple replicates of the same sample are of the same datatype". And no, this test on replicates was not in dev when I created the PR, but only libraries were tested.

File: bin/check_samplesheet.py (hunk 2 above)
Comment (drpatelh) [reply bin/check_samplesheet.py:141]: Well spotted! 🚀

**Comment by drpatelh** [comment]
I have cherry picked the changes and added in https://github.com/nf-core/atacseq/pull/243/commits/cb4f53f3ba7cca49b3f76b1d36ca26d1a6ef1e4e 

Do those changes look ok to you guys in which case we can ignore the failing tests here.
